### ISSUE TITLE
Experiencing a weird formatting bug with bigint

### ISSUE THREAD

**Issue by ch99q** [oven-sh/bun#335]
So I was trying to run a performance test on my [decimal precision util](https://gist.github.com/ca5886ab63ac16d2963f708d492bd359) and came across a formatting issue, the value is correct, but it says the BigInt has decimals for some reason.

<img width="481" alt="Screenshot 2022-07-07 at 09 21 53" src="https://user-images.githubusercontent.com/42772328/177716051-45f647d3-5745-48f6-beca-d4581eab4bb4.png">

But daymn impressive performance! 🚀

**Comment by Jarred-Sumner** [comment]
Thanks for reporting. This is definitely a bug

**Comment by imskr** [comment]
Can I work on this?

**Comment by Jarred-Sumner** [comment]
That’d be great. I believe the bug is in exports.zig or bindings.zig, probably where it tries to figure out how to format console.log. If you grep for BigInt, filtered to .zig files it’ll probably show up

**Comment by ch99q** [comment]
Did some more digging and a simple `console.log(0 - 0.001);` outputs `-0.001n`

For some reason, when bun detects a decimal place, it writes it as a BigInt. Regular integers work fine.

**Comment by bombitmanbomb** [comment]
It seems to format as BigInt if the number is Negative or has a decimal place.

**Comment by jhmaster2000** [comment]
More accurately, bun seems to be formatting as BigInt anything that is non-integer or outside of the Int32 range (-2147483648 to 2147483647), not sure why though
```js
console.log(2147483647); // 2147483647
console.log(2147483648); // 2147483648n
console.log(-2147483648); // -2147483648
console.log(-2147483649); // -2147483649n
console.log(7.0); // 7
console.log(7.01); // 7.01n
console.log(-7.0); // -7
console.log(-7.01); // -7.01n
```

**Comment by evanwashere** [comment]
fixed in #509 #508
